### PR TITLE
Fix subtitles won't show, any source, #3431

### PR DESCRIPTION
**Pull request by low-batt**
This commit will:

- Change OpenSubSubtitle.hash to throw noResult if the video is being streamed
- Change OpenSubSubtitle.requestIMDB to use the media title as the text for the lookup if the video is being streamed
- Change OpenSubSubtitle.hash and ShooterSubtitle.hash to ensure file is closed
- Improve logging when obtaining subtitles
- Change MainMenuActions.saveDownloadedSub to bring up a Save panel to ask the user where to save the downloaded subtitle file
- Change the "Save Downloaded Subtitle" menu item to end with an ellipsis to indicate the user will be prompted for additional information
- Change Utility.quickSavePanel to accept an optional filename parameter
- Add "Cannot connect" OSD message
- Add "Timed out" OSD message
- Change subtitle classes to use an extension when defining the logger subsystem
- Add parameter labels to trailing closures in Assrt methods

With these changes subtitles can be downloaded and viewed for movies streamed from YouTube and from Dailymotion when using OpenSubtitles.org.

This is a reimplementation of the fix in PR #3499 that was applied only to the 1.3.0 branch to avoid merge conflicts with the plugin system. The fix has been adapted to be compatible with the changes made to the subtitle modules to support the plugin system.

The reimplementation also adds two new OSD messages for common failures when connecting to opensubtitles.org and improves some of the logging.

This commit also eliminates Swift compiler build warnings due to backward matching of unlabeled trailing closure being deprecated. These warnings had been eliminated but were reintroduced during the large merge of the plugin system.

- [ ] This change has been discussed with the author.
- [x] It implements / fixes issue #3431.

---

**Description:**
A version of this fix was merged into the 1.3.0 branch but not into the develop branch. Now that the plugin system has been merged into the develop branch the fix has been reimplemented in this PR to be compatible with the current subtitle system.

### CODE REVIEW

File: iina/Base.lproj/MainMenu.xib [hunk 1]
```diff
@@ -551,7 +551,7 @@ CA
                                 <modifierMask key="keyEquivalentModifierMask"/>
                                 <menu key="submenu" title="Find Online Subtitles from" id="i0H-bE-bH8"/>
                             </menuItem>
-                            <menuItem title="Save Downloaded Subtitle" id="9zk-co-roT">
+                            <menuItem title="Save Downloaded Subtitle..." id="9zk-co-roT">
```
Comment (uiryuu) [inline iina/Base.lproj/MainMenu.xib:554]: Use ellipsis (…)

File: iina/Base.lproj/Localizable.strings [hunk 2]
```diff
@@ -80,6 +80,8 @@
 "osd.file_error" = "Error reading file";
 "osd.cannot_login" = "Cannot login";
 "osd.canceled" = "Canceled";
+"osd.cannotConnect" = "Cannot connect";
```
Comment (uiryuu) [inline iina/Base.lproj/Localizable.strings:83]: `cannot_connect`

File: iina/Base.lproj/Localizable.strings [hunk 3]
```diff
@@ -80,6 +80,8 @@
...
+"osd.timedOut" = "Timed Out";
```
Comment (uiryuu) [inline iina/Base.lproj/Localizable.strings:84]: `timed_out`

**Review by uiryuu — CHANGES_REQUESTED** [review]
IMO the best practice for localization is to change the base localization only and push to Crowdin. It might cause chaos if we sync dev branch and Crowdin back and forth.

**Comment by low-batt** [comment]
Yes, this PR is should not contain the language translations. The original PR was created before I was informed of the proper practice. Since that one was merged into the 1.3.0 branch I included the translations in this one. Let me know if you would like me to change the PR to not include the translations.

I'll update the PR with the other fixes today.

File: iina/Base.lproj/MainMenu.xib (hunk 1 above)
Comment (low-batt) [reply iina/Base.lproj/MainMenu.xib]: Fixed in new commit.
Thanks for catching!

File: iina/Base.lproj/Localizable.strings (hunk 2 above)
Comment (low-batt) [reply iina/Base.lproj/Localizable.strings]: Fixed in new commit.
Thanks for catching!

File: iina/Base.lproj/Localizable.strings (hunk 3 above)
Comment (low-batt) [reply iina/Base.lproj/Localizable.strings]: Fixed in new commit.
Thanks for catching!

**Comment by lhc70000** [comment]
Although I think it's fine to include translations, now we have a conflict after merging those from Crowdin. Could you remove the translation changes (at least for pl.lproj)?

**Comment by low-batt** [comment]
I rebased with develop and fixed the merge conflict.

**Comment by uiryuu** [comment]
Can you remove all the translations?

**Comment by uiryuu** [comment]
> Happy to remove all the translations, but please confirm you really want me to do that.

Yes, please remove them.

**Comment by low-batt** [comment]
I will remove the translations. I should be able to get that done tomorrow.

**Comment by low-batt** [comment]
Closing this PR in favor of the new PR #3975 which only contains the base language translation files.

**Pull request closed without merging**